avalon/tools/cbsceneinventory/app.py

```diff
@@ -101,8 +101,53 @@ def _on_update_to_latest(items):
         menu.addAction(expandall_action)
         menu.addAction(collapse_action)
 
+        custom_actions = self.get_custom_actions(containers=items)
+        if custom_actions:
+            menu.addSeparator()
+            submenu = QtWidgets.QMenu("Actions", self)
+            for Action in custom_actions:
+
+                color = Action.color or DEFAULT_COLOR
+                icon = qta.icon("fa.%s" % Action.icon, color=color)
+                action_item = QtWidgets.QAction(icon, Action.label, submenu)
+                action_item.triggered.connect(
+                    partial(self.process_custom_action, Action, items))
```

Comment:
There's a slight "logic" issue here, the items should actually be containers and *NOT!* include the additional metadata that the inventory manager has inserted into the item, because the action is build to run over the `container` and not the `item`.

Maybe the inventory manager's model should hold a `item["container"]` key that keeps a copy of the original `container` data? Then that can be used here to process containers.

Reply:
I know we already protect the container by adding it to  the custom data but the reversed logic could also be applied where we explicitly say `container["metadata"] = custom_data`. This would make more sense when you look at the function: `collect_container_metadata`.

I think it would make the entire logic more secure either way.